resources/js/app/components/permission/permission.vue

```diff
@@ -0,0 +1,41 @@
+<template>
+    <span v-if="hasPermission()" v-title="title()">
+        <Icon icon="carbon:locked" v-if="isLocked()"></Icon>
+        <Icon icon="carbon:unlocked" v-if="!isLocked()"></Icon>
+        <Icon icon="carbon:tree-view" v-if="inherited"></Icon>
+    </span>
+</template>
+
+<script>
+import { t } from 'ttag';
+
+export default {
+
+    name: 'permission',
+
+    props: {
+        inherited: false,
+        objectRoles: [],
+        role: '',
+        userRoles: [],
```

Comment:
```suggestion
        inherited: {
            type: Boolean,
            default: false,
        },
        objectRoles: {
            type: Array,
            default: () => ([]),
        },
        role: {
            type: String,
            default: '',
        },
        userRoles: {
            type: Array,
            default: () => ([]),
        },
```

When defining `props` as an object, keys are prop name and values are prop types; if we want to define defaults each prop must be defined as an object with `type` and `default` properties.
Furthermore, defaults of props that are array or objects must be returned from a factory function.